flow/project.py

```diff
@@ -557,6 +557,9 @@ def __init__(self, condition, tag=None):
                 super(pre, self).__init__(condition, tag)
 
             def __call__(self, func):
+                for operation_i in self._parent_class._OPERATION_FUNCTIONS:
```

Comment:
Since `_OPERATION_FUNCTIONS` variable is currently a list of tuples of the form (`name`, `func`), it's cleaner to loop over the tuples directly and check equality with the second component:

```suggestion
                for (name, func) in self._parent_class._OPERATION_FUNCTIONS:
```

and then check if self.condition is equal to func.